src/main/java/edu/harvard/iq/dataverse/DataverseServiceBean.java

```diff
@@ -91,21 +92,33 @@ public Dataverse save(Dataverse dataverse) {
        
         dataverse.setModificationTime(new Timestamp(new Date().getTime()));
         Dataverse savedDataverse = em.merge(dataverse);
+        return savedDataverse;
+    }
+    
+    public boolean index(Dataverse dataverse) {
+        return index(dataverse, false);
+
+    }
+        
+    public boolean index(Dataverse dataverse, boolean indexPermissions) {    
         /**
          * @todo check the result to see if indexing was successful or not
          * added logging of exceptions 
          */
         try {
-            Future<String> indexingResult = indexService.indexDataverse(savedDataverse);
+            indexService.indexDataverse(dataverse);
+            if (indexPermissions) {
+                solrIndexService.indexPermissionsOnSelfAndChildren(dataverse);
+            }
         } catch (IOException | SolrServerException e) {
-            String failureLogText = "Post-save indexing failed. You can kickoff a re-index of this dataverse with: \r\n curl http://localhost:8080/api/admin/index/dataverses/" + savedDataverse.getId().toString();
+            String failureLogText = "Post-save indexing failed. You can kickoff a re-index of this dataverse with: \r\n curl http://localhost:8080/api/admin/index/dataverses/" + dataverse.getId().toString();
             failureLogText += "\r\n" + e.getLocalizedMessage();
-            LoggingUtil.writeOnSuccessFailureLog(null, failureLogText, savedDataverse);
+            LoggingUtil.writeOnSuccessFailureLog(null, failureLogText, dataverse);
```

Comment:
Kind of a nit, but there are cases where this isn't being called by onSuccess so the failure message so maybe add to the general log in those cases.

Reply:
This is a brand new method and all the calls I added are in onSuccess methods. Additionally as a tech design decision, we've said we want all such indexing to be onSuccess, no? So I think it's ok.

Note, that currently not all indexing is actually on success (see MoveDataverse), but changing that was out of scope for this issue).